ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOPipelineModel.scala

```diff
@@ -124,11 +124,12 @@ class H2OMOJOPipelineModel(override val uid: String)
   }
 
   override def transform(dataset: Dataset[_]): DataFrame = {
-    val flatten = H2OSchemaUtils.flattenDataFrame(dataset.toDF())
-    val names = flatten.schema.fields.map(f => flatten(f.name))
+    val flattenedDF = H2OSchemaUtils.flattenDataFrame(dataset.toDF())
+    val relevantColumnNames = flattenedDF.columns.intersect(getInputNames())
```

Comment:
@mn-mikke Please don't use input names here.

Correct approach is to define featureCols and define them during creating of H2OMOjoModel. Here, you should use getFeaturesCols.

GetInputNames will get removed in the future.